Apps/WebClient/src/ClientApp/src/components/report/ReportsComponent.vue

```diff
@@ -0,0 +1,626 @@
+import { ReportFormatType } from "@/models/reportRequest";
+import RequestResult from "@/models/requestResult";
+import container from "@/plugins/container";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import { ILogger } from "@/services/interfaces";
+import ConfigUtil from "@/utility/configUtil";
+import EventTracker from "@/utility/eventTracker";
+
+const medicationReport = "medication-report";
+const mspVisitReport = "msp-visit-report";
+const covid19Report = "covid19-report";
+const immunizationReport = "immunization-report";
+const medicationRequestReport = "medication-request-report";
+const noteReport = "note-report";
+const laboratoryReport = "laboratory-report";
+const hospitalVisitReport = "hospital-visit-report";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const options: any = {
+    components: {
+        LoadingComponent,
+        "message-modal": MessageModalComponent,
+        medicationReport: MedicationHistoryReportComponent,
+        mspVisitReport: MSPVisitsReportComponent,
+        covid19Report: Covid19ReportComponent,
+        immunizationReport: ImmunizationHistoryReportComponent,
+        medicationRequestReport: MedicationRequestReportComponent,
+        "hg-date-picker": DatePickerComponent,
+        MultiSelectComponent,
+        noteReport: NotesReportComponent,
+        laboratoryReport: LaboratoryReportComponent,
+        HospitalVisitReport: HospitalVisitReportComponent,
+    },
+};
+
+@Component(options)
+export default class ReportsComponent extends Vue {
+    @Prop({ required: true })
+    hdid!: string;
+
+    @Getter("webClient", { namespace: "config" })
+    config!: WebClientConfiguration;
+
+    @Getter("laboratoryOrdersAreQueued", { namespace: "laboratory" })
+    laboratoryOrdersAreQueued!: (hdid: string) => boolean;
+
+    @Getter("medications", { namespace: "medication" })
+    medications!: (hdid: string) => MedicationStatementHistory[];
+
+    @Getter("patientData", { namespace: "user" })
+    patientData!: PatientData;
+
+    @Ref("messageModal")
+    readonly messageModal!: MessageModalComponent;
+
+    @Ref("report")
+    readonly report!: {
+        generateReport: (
+            reportFormatType: ReportFormatType,
+            headerData: ReportHeader
+        ) => Promise<RequestResult<Report>>;
+    };
+
+    @Action("addError", { namespace: "errorBanner" })
+    addError!: (params: {
+        errorType: ErrorType;
+        source: ErrorSourceType;
+        traceId: string | undefined;
+    }) => void;
+
+    @Action("setTooManyRequestsError", { namespace: "errorBanner" })
+    setTooManyRequestsError!: (params: { key: string }) => void;
+
+    private ReportFormatType: unknown = ReportFormatType;
+    private isLoading = false;
+    private isGeneratingReport = false;
+    private reportFormatType = ReportFormatType.PDF;
+    private reportComponentName = "";
+    private reportTypeOptions = [{ value: "", text: "Select" }];
```

Comment:
You may as well remove the `private` modifiers from the properties and getters in the component while you are in here to satisfy Volar.